src/pages/Fox/hooks/useFoxPageContext.tsx

```diff
@@ -0,0 +1,64 @@
+import { type AccountId, type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import React, { createContext, useContext, useMemo, useState } from 'react'
+import {
+  selectAccountIdByAccountNumberAndChainId,
+  selectAccountNumberByAccountId,
+} from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
+
+type FoxPageContextType = {
+  selectedAssetAccountId: AccountId | undefined
+  assetId: AssetId
+  setAssetAccountId: React.Dispatch<React.SetStateAction<AccountId | undefined>>
+  assetAccountNumber: number | undefined
+}
+
+const FoxPageContext = createContext<FoxPageContextType | undefined>(undefined)
+
+export const FoxPageProvider: React.FC<React.PropsWithChildren<{ assetId: AssetId }>> = ({
+  children,
+  assetId,
+}) => {
+  const [assetAccountId, setAssetAccountId] = useState<AccountId | undefined>()
+
+  const filter = useMemo(
+    () => (assetAccountId && assetId ? { assetId, accountId: assetAccountId } : undefined),
+    [assetAccountId, assetId],
+  )
+
+  const assetAccountNumber = useAppSelector(state =>
+    filter ? selectAccountNumberByAccountId(state, filter) : undefined,
+  )
+
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdByAccountNumberAndChainId,
+  )
+
+  const selectedAssetAccountId = useMemo(() => {
+    if (!(filter && assetAccountNumber !== undefined)) return
+    const accountNumberAccountIds = accountIdsByAccountNumberAndChainId[assetAccountNumber]
+    const matchingAccountId = accountNumberAccountIds?.[fromAssetId(assetId).chainId]
+    return matchingAccountId
+  }, [accountIdsByAccountNumberAndChainId, filter, assetId, assetAccountNumber])
+
+  const value: FoxPageContextType = useMemo(
+    () => ({
+      selectedAssetAccountId,
+      setAssetAccountId,
+      assetId,
+      assetAccountId,
```

Comment:
Unused and not present in `FoxPageContextType`, remove me

Reply:
Came back after the next comment in [e05e4b2](https://github.com/shapeshift/web/pull/7882/commits/e05e4b244591c3a87f0da0fca531273c9a08a4c3)